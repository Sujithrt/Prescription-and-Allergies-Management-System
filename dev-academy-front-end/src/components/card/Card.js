import React from 'react';
import Box from '@mui/material/Box';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import Button from 'react-bootstrap/Button';
import Typography from '@mui/material/Typography';
import './../card/card.css';
import { useHistory } from 'react-router-dom';

/**
 * Displays Patient Details and allows User to click a button to get more patient details
 * @param {Patient Details} props 
 * @returns 
 */
export default function OutlinedCard(props) {

    const history = useHistory();

    return (
        <Box sx={{ minWidth: 275 }} className="box">
            <Card variant="outlined" className="cards">
                <React.Fragment>
                    <CardContent>
                        <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
                            Registration Number - {props.patient.id}<br />
                        </Typography>
                        <Typography variant="h5">
                            {props.patient.firstName} {props.patient.lastName}
                        </Typography>
                        <Typography variant="body2">
                            Age: {props.patient.age}<br /><br />
                            Gender: {props.patient.gender}<br />
                        </Typography>
                        <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
                            Last Encounter Date: {props.patient.date}<br />
                        </Typography>
                    </CardContent>
                    <span className="button-align"><CardActions><br /><br />
                        <Button variant="primary" onClick={() => history.push(`/patient/${props.patient.id}`)}>More Details</Button>
                    </CardActions></span>
                </React.Fragment>
            </Card>
        </Box>
    );
}