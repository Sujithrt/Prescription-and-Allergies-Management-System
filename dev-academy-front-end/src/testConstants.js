export const patient = {
    "id": 1001,
    "firstName": "Leanne",
    "lastName": "Graham",
    "gender": "Male",
    "age": 21,
    "phoneNumber": 9123456780,
    "email": "leanne.graham@gmail.com",
    "city": "Bangalore",
    "state": "Karnataka",
    "medicalConditions": [
        {
            "id": 1,
            "name": "Cephalalgia (headache)"
        },
        {
            "id": 3,
            "name": "Otitis externa (swimmer's ear)"
        },
        {
            "id": 9,
            "name": "Contusion (bruise)"
        }
    ],
    "allergies": [
        {
            "id": 7,
            "name": "Aspirin Allergy",
            "reaction": "Tiredness",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 4,
                "name": "Aspirin"
            },
            "encounter": {
                "id": 10,
                "date": "2021-10-26",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 8,
            "name": "Paracetamol",
            "reaction": "sdfd",
            "severity": "MILD",
            "status": "ACTIVE",
            "ingredient": {
                "id": 1,
                "name": "Paracetamol"
            },
            "encounter": {
                "id": 13,
                "date": "2021-10-26",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 13,
            "name": "Ibuprofin Allergy",
            "reaction": "Increased Perspiration",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 3,
                "name": "Ibuprofen"
            },
            "encounter": {
                "id": 20,
                "date": "2021-10-29",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 14,
            "name": "Paracetamol Allergy",
            "reaction": "Increased Perspiration",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 3,
                "name": "Ibuprofen"
            },
            "encounter": {
                "id": 1,
                "date": "2021-10-26",
                "patient": {
                    "id": 1003,
                    "firstName": "Clementine",
                    "lastName": "Bauch",
                    "gender": "Female",
                    "age": 56,
                    "phoneNumber": 9123456782,
                    "email": "clementine.bauch@gmail.com",
                    "city": "Mysore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 23,
                            "name": "Gestation (pregnancy) "
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        }
    ]
};

export const user = {
    "id": 1,
    "username": "John",
    "email": "john.cage@gmail.com",
    "role": "Physician",
    "firstName": "John",
    "lastName": "Cage",
    "speciality": "Cardiologist",
    "city": "Bangalore",
    "state": "Karnataka",
    "dob": "1975-06-12T18:30:00.000+00:00",
    "authorities": [
        {
            "id": 1,
            "roleCode": "USER",
            "roleDescription": "User role",
            "authority": "USER"
        },
        {
            "id": 2,
            "roleCode": "ADMIN",
            "roleDescription": "Admin role",
            "authority": "ADMIN"
        }
    ]
};

export const prescription = [
    {
        "id": {
            "encounterId": 1,
            "medicineId": 1
        },
        "encounter": {
            "id": 29,
            "date": "2021-11-07",
            "patient": {
                "id": 1009,
                "firstName": "Glenna",
                "lastName": "Reichert",
                "gender": "Female",
                "age": 65,
                "phoneNumber": 9123456788,
                "email": "glenna.reichert@gmail.com",
                "city": "Kolkata",
                "state": "West Bengal",
                "medicalConditions": [
                    {
                        "id": 6,
                        "name": "Tussis (cough)"
                    },
                    {
                        "id": 7,
                        "name": "Pyrexia (fever)"
                    }
                ]
            },
            "user": {
                "id": 1,
                "username": "John",
                "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                "email": "john.cage@gmail.com",
                "enabled": true,
                "role": "Physician",
                "authorities": [
                    {
                        "id": 1,
                        "roleCode": "USER",
                        "roleDescription": "User role",
                        "authority": "USER"
                    },
                    {
                        "id": 2,
                        "roleCode": "ADMIN",
                        "roleDescription": "Admin role",
                        "authority": "ADMIN"
                    }
                ],
                "credentialsNonExpired": true,
                "accountNonExpired": true,
                "accountNonLocked": true
            }
        },
        "medicine": {
            "id": 1,
            "genericName": "Paracetamol",
            "brandName": "Dolo",
            "category": {
                "id": 1,
                "name": "Antipyretics"
            },
            "ingredients": [
                {
                    "id": 1,
                    "name": "Paracetamol"
                }
            ]
        },
        "dosage": "1 time a day"
    }
];

export const allergy = {
    "id": 19,
    "name": "Paracetamol Allergy",
    "reaction": "Tiredness",
    "severity": "MILD",
    "status": "ACTIVE",
    "ingredient": {
        "id": 1,
        "name": "Paracetamol"
    },
    "encounter": {
        "id": 28,
        "date": "2021-11-07",
        "patient": {
            "id": 1009,
            "firstName": "Glenna",
            "lastName": "Reichert",
            "gender": "Female",
            "age": 65,
            "phoneNumber": 9123456788,
            "email": "glenna.reichert@gmail.com",
            "city": "Kolkata",
            "state": "West Bengal",
            "medicalConditions": [
                {
                    "id": 6,
                    "name": "Tussis (cough)"
                },
                {
                    "id": 7,
                    "name": "Pyrexia (fever)"
                }
            ]
        },
        "user": {
            "id": 1,
            "username": "John",
            "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
            "email": "john.cage@gmail.com",
            "enabled": true,
            "role": "Physician",
            "authorities": [
                {
                    "id": 1,
                    "roleCode": "USER",
                    "roleDescription": "User role",
                    "authority": "USER"
                },
                {
                    "id": 2,
                    "roleCode": "ADMIN",
                    "roleDescription": "Admin role",
                    "authority": "ADMIN"
                }
            ],
            "credentialsNonExpired": true,
            "accountNonExpired": true,
            "accountNonLocked": true
        }
    },
    "patient": {
        "id": 1009,
        "firstName": "Glenna",
        "lastName": "Reichert",
        "gender": "Female",
        "age": 65,
        "phoneNumber": 9123456788,
        "email": "glenna.reichert@gmail.com",
        "city": "Kolkata",
        "state": "West Bengal",
        "medicalConditions": [
            {
                "id": 6,
                "name": "Tussis (cough)"
            },
            {
                "id": 7,
                "name": "Pyrexia (fever)"
            }
        ]
    }
};

export const recentPatients = [
    {
        "id": 1009,
        "firstName": "Glenna",
        "lastName": "Reichert",
        "age": 65,
        "gender": "Female",
        "date": "2021-11-09"
    },
    {
        "id": 1008,
        "firstName": "Nicholas",
        "lastName": "Runolfsdottir",
        "age": 21,
        "gender": "Male",
        "date": "2021-11-07"
    },
    {
        "id": 1021,
        "firstName": "Leroy",
        "lastName": "Blair",
        "age": 21,
        "gender": "Male",
        "date": "2021-11-07"
    },
    {
        "id": 1027,
        "firstName": "Alanna",
        "lastName": "Cantrell",
        "age": 65,
        "gender": "Female",
        "date": "2021-11-07"
    },
    {
        "id": 1016,
        "firstName": "Osbert",
        "lastName": "Valdez",
        "age": 32,
        "gender": "Male",
        "date": "2021-11-02"
    },
    {
        "id": 1024,
        "firstName": "Emmett",
        "lastName": "Marshall",
        "age": 45,
        "gender": "Male",
        "date": "2021-11-02"
    }
];

export const prescriptionData = [
    {
        "brandName": "Tylenol",
        "genericName": "Acetaminophen",
        "dosage": "3 times a day",
        "date": "2021-10-26"
    },
    {
        "brandName": "Advil",
        "genericName": "Ibuprofen",
        "dosage": "3 times a day",
        "date": "2021-10-26"
    },
    {
        "brandName": "Dolo",
        "genericName": "Paracetamol",
        "dosage": "1 time a day",
        "date": "2021-10-26"
    }
];

export const medicineList = [
    {
        "id": 1,
        "brandName": "Dolo",
        "cause": "Incompatible with patients' medical conditions"
    },
    {
        "id": 13,
        "brandName": "Vibramycin",
        "cause": ""
    },
    {
        "id": 17,
        "brandName": "Vibramycin",
        "cause": ""
    },
    {
        "id": 22,
        "brandName": "Betadine",
        "cause": ""
    },
    {
        "id": 25,
        "brandName": "Iodex",
        "cause": ""
    },
    {
        "id": 46,
        "brandName": "Desyrel",
        "cause": "Incompatible with patients' medical conditions"
    }
];