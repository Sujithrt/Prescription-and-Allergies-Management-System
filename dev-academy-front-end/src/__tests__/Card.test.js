import React from 'react';
import { shallowToJson } from "enzyme-to-json";
import { shallow, mount } from 'enzyme';
import Card from '../components/card/Card'
import { Router } from 'react-router-dom';
import { act } from '@testing-library/react-hooks';
import { patient } from '../testConstants';


describe("Card Test", () => {
    it("Card is rendered as expected", () => {
        const card = shallow(<Card patient={patient} />);
        expect(shallowToJson(card)).toMatchSnapshot();
    });
});

describe('Card More details button click', () => {
    it('Redirects to correct URL on click', () => {
        const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
        const wrapper = mount(
            <Router history={historyMock}>
                <Card patient={patient} />
            </Router>,
        ).find('button')

        const { onClick } = wrapper.props();
        act(() => {
            onClick();
        });
        expect(historyMock.push.mock.calls[0][0]).toEqual(`/patient/${patient.id}`);
    });
})