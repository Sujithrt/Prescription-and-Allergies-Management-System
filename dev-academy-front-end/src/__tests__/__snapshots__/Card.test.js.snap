// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test Card is rendered as expected 1`] = `
<ForwardRef(Box)
  className="box"
  sx={
    Object {
      "minWidth": 275,
    }
  }
>
  <ForwardRef(Card)
    className="cards"
    variant="outlined"
  >
    <ForwardRef(CardContent)>
      <ForwardRef(Typography)
        color="text.secondary"
        gutterBottom={true}
        sx={
          Object {
            "fontSize": 14,
          }
        }
      >
        Registration Number - 
        1001
        <br />
      </ForwardRef(Typography)>
      <ForwardRef(Typography)
        variant="h5"
      >
        Leanne
         
        Graham
      </ForwardRef(Typography)>
      <ForwardRef(Typography)
        variant="body2"
      >
        Age: 
        21
        <br />
        <br />
        Gender: 
        Male
        <br />
      </ForwardRef(Typography)>
      <ForwardRef(Typography)
        color="text.secondary"
        gutterBottom={true}
        sx={
          Object {
            "fontSize": 14,
          }
        }
      >
        Last Encounter Date: 
        <br />
      </ForwardRef(Typography)>
    </ForwardRef(CardContent)>
    <span
      className="button-align"
    >
      <ForwardRef(CardActions)>
        <br />
        <br />
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          variant="primary"
        >
          More Details
        </Button>
      </ForwardRef(CardActions)>
    </span>
  </ForwardRef(Card)>
</ForwardRef(Box)>
`;
